metrix-integration/src/main/java/com/powsybl/metrix/integration/MetrixNetwork.java

```diff
@@ -317,37 +314,40 @@ private void createLineList() {
         }
     }
 
-    private void createTransformerList() {
-        int nbNok = 0;
-        int nbPtcNok = 0;
-
-        for (TwoWindingsTransformer twt : network.getTwoWindingsTransformers()) {
-            Terminal t1 = twt.getTerminal1();
-            Terminal t2 = twt.getTerminal2();
-            Bus b1 = t1.getBusBreakerView().getBus();
-            Bus b2 = t2.getBusBreakerView().getBus();
-            if (b1 != null && b2 != null) {
-                if (busList.contains(b1) && busList.contains(b2)) {
-                    addTwoWindingsTransformer(twt);
-                    if (twt.hasPhaseTapChanger()) {
-                        addPhaseTapChanger(twt);
-                    }
-                } else {
-                    nbNok++;
-                    if (twt.hasPhaseTapChanger()) {
-                        nbPtcNok++;
-                    }
+    private void checkTwoWindingsTransformer(TwoWindingsTransformer twt, AtomicInteger nbNok, AtomicInteger nbPtcNok) {
+        Terminal t1 = twt.getTerminal1();
+        Terminal t2 = twt.getTerminal2();
+        Bus b1 = t1.getBusBreakerView().getBus();
+        Bus b2 = t2.getBusBreakerView().getBus();
+        if (b1 != null && b2 != null) {
+            if (busList.contains(b1) && busList.contains(b2)) {
+                addTwoWindingsTransformer(twt);
+                if (twt.hasPhaseTapChanger()) {
+                    addPhaseTapChanger(twt);
                 }
             } else {
-                nbNok++;
+                nbNok.incrementAndGet();
+                if (twt.hasPhaseTapChanger()) {
+                    nbPtcNok.incrementAndGet();
+                }
             }
+        } else {
+            nbNok.incrementAndGet();
         }
+    }
+
+    private void createTwoWindingsTransformersList() {
+        AtomicInteger nbNok = new AtomicInteger(0);
+        AtomicInteger nbPtcNok = new AtomicInteger(0);
+        network.getTwoWindingsTransformers().forEach(twt -> checkTwoWindingsTransformer(twt, nbNok, nbPtcNok));
         if (LOGGER.isDebugEnabled()) {
-            LOGGER.debug(String.format("Twotrfo    total = <%5d> ok = <%5d> not = <%5d>", twoWindingsTransformerList.size() + nbNok, twoWindingsTransformerList.size(), nbNok));
-            LOGGER.debug(String.format("PhaseTC    total = <%5d> ok = <%5d> not = <%5d>", phaseTapChangerList.size() + nbPtcNok, phaseTapChangerList.size(), nbPtcNok));
+            LOGGER.debug(String.format("Twotrfo    total = <%5d> ok = <%5d> not = <%5d>", twoWindingsTransformerList.size() + nbNok.get(), twoWindingsTransformerList.size(), nbNok.get()));
+            LOGGER.debug(String.format("PhaseTC    total = <%5d> ok = <%5d> not = <%5d>", phaseTapChangerList.size() + nbPtcNok.get(), phaseTapChangerList.size(), nbPtcNok.get()));
         }
+    }
 
-        nbNok = 0;
+    private void createThreeWindingsTransformersList() {
```

Comment:
I just extracted this method: if we want to remove it (which I'm not in favour of since the exception is thrown elsewhere and I'd prefer to add the functionality), it should be done in another dedicated PR since it's not related to the Sonar issues